src/charm.py

```diff
@@ -392,21 +390,36 @@ def _config_changed(self, event: HookEvent) -> None:
         # First execute the setup script in 2 conditions:
         # - First run (when no services are planned in pebble)
         # - Change in important S3 parameter (comparing value with envVars in pebble plan)
-        if (
-            self._is_config_valid()
-            and self.model.unit.is_leader()
-            and self._should_run_setup(current_plan, previous_s3_info)
-        ):
+        if self._is_config_valid() and self.model.unit.is_leader():
+            env_settings = self._create_discourse_environment_settings()
             try:
-                self.model.unit.status = MaintenanceStatus("Compiling assets")
-                script = f"{SCRIPT_PATH}/pod_setup.sh"
-                process = container.exec(
-                    [script],
-                    environment=self._create_discourse_environment_settings(),
-                    working_dir="/srv/discourse/app",
-                    user="discourse",
-                )
-                process.wait_output()
+                if not current_plan.services:
+                    self.model.unit.status = MaintenanceStatus("Compiling assets")
+                    script = f"{SCRIPT_PATH}/pod_setup.sh"
+                    process = container.exec(
+                        [script],
+                        environment=env_settings,
+                        working_dir="/srv/discourse/app",
+                        user="discourse",
+                    )
+                    process.wait_output()
+                    self.model.unit.status = MaintenanceStatus("Compiling assets")
```

Comment:
This status already seems to be set on line 397?

Reply:
Fixed